pkg/utils/nsenter/nsenter.go

```diff
@@ -51,3 +54,37 @@ func NetnsEnter(pid int, f func() error) error {
 
 	return f()
 }
+
+func CgroupnsEnter(pid int, f func() error) error {
+	if pid == 0 {
+		return f()
+	}
+
+	// Lock the OS Thread so we don't accidentally switch namespaces
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	// Save the current cgroup namespace
+	origCgroupPath := fmt.Sprintf("/proc/%d/task/%d/ns/cgroup", os.Getpid(), unix.Gettid())
+	origns, err := unix.Open(origCgroupPath, unix.O_RDONLY|unix.O_CLOEXEC, 0)
+	if err != nil {
+		return err
+	}
+	defer unix.Close(origns)
+
+	newCgroupPath := fmt.Sprintf("%s/%d/ns/cgroup", host.HostProcFs, pid)
```

Comment:
I can use `filepath.Join`.

But not `SecureJoin()` because `host.HostProcFs` is not a root (e.g. `/host` or `/proc/PID/root`) but a procfs (`/host/proc`). It would not make sense to interpret the symlinks relative to the procfs. In any case,  `/host/proc` is trusted here, and there are no symlinks to interpret relative to a root, so we don't need `SecureJoin()`.

Reply:
Is there really any advantage of using filepath.Join()? I see the netns package uses `fmt.Sprintf("/proc/%d/task/%d/ns/net", pid, tid)` and I find it more readable.

Reply:
I also find `Sprintf` more readable. I think here we can use either.

This code is Linux-only so we don't need to support Windows-style paths (with path separator `\` and volume prefix `C:`). And the path is already clean so we don't need the call to `filepath.Clean`.

`filepath.Join()` has the small advantage that we don't need to think about Windows support.